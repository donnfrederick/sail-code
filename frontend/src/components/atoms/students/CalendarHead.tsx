import moment from 'moment'
import * as React from 'react'
import styled from 'styled-components'
import resolvePath from 'utils/resolvePath'
import zeroPadding from 'utils/zeroPadding'

interface Props {
  authToken: string
  year: number
  month: number
  changeYear(year: number): void
  changeMonth(month: number): void
  getCalendar(authToken: string, year: number, month: number): void
  resetRecommendedReservations(): void
}

export default (props: Props) => {
  const {
    authToken,
    year,
    month,
    changeYear,
    changeMonth,
    getCalendar,
    resetRecommendedReservations
  } = props

  return (
    <Container>
      <Month>
        {moment(`${year}-${zeroPadding(month)}`).format('MMMM YYYY')}
      </Month>
      <LeftArrow
        src={resolvePath.image('common/left-arrow.png')}
        data-disabled={
          moment().diff(moment(`${year}-${zeroPadding(month + 1)}`), 'month') >=
          0
        }
        onClick={() => {
          if (month <= 1) {
            changeYear(year - 1)
            changeMonth(12)
            getCalendar(authToken, year - 1, 12)
          } else {
            changeMonth(month - 1)
            getCalendar(authToken, year, month - 1)
          }
          resetRecommendedReservations()
        }}
      />
      <RightArrow
        src={resolvePath.image('common/right-arrow.png')}
        onClick={() => {
          if (month >= 12) {
            changeYear(year + 1)
            changeMonth(1)
            getCalendar(authToken, year + 1, 1)
          } else {
            changeMonth(month + 1)
            getCalendar(authToken, year, month + 1)
          }
          resetRecommendedReservations()
        }}
      />
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  width: 100%;
  height: 112px;
  box-sizing: border-box;
  border-bottom: 1px solid #cad5db;
`

const Month = styled.div`
  font-size: 40px;
  font-weight: 500;
  letter-spacing: 0.5px;
  text-align: center;
  color: #405766;
`

const LeftArrow = styled.img`
  position: absolute;
  left: 48px;
  bottom: 32px;
  width: 20px;
  height: 34px;

  &[data-disabled='true'] {
    pointer-events: none;
    opacity: 0.4;
  }
`

const RightArrow = styled.img`
  position: absolute;
  right: 48px;
  bottom: 32px;
  width: 20px;
  height: 34px;

  &[data-disabled='true'] {
    pointer-events: none;
    opacity: 0.4;
  }
`
