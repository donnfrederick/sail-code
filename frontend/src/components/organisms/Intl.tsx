import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
