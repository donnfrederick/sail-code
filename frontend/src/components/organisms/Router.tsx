import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
