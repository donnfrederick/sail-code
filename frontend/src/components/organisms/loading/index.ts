import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
