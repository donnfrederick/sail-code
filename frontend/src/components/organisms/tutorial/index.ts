import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
