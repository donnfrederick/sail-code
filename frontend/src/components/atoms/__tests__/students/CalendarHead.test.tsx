import CalendarHead from 'components/atoms/students/CalendarHead'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <FooterButton />', () => {
  const tree = renderer
    .create(
      <Intl>
        <CalendarHead
          authToken={''}
          year={1970}
          month={1}
          changeYear={() => null}
          changeMonth={() => null}
          getCalendar={() => null}
          resetRecommendedReservations={() => null}
        />
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
