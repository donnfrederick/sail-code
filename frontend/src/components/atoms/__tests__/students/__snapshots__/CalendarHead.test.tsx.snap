// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <FooterButton /> 1`] = `
<div
  className="sc-bdVaJa hhYVyr"
>
  <div
    className="sc-bwzfXH jEtEnA"
  >
    January 1970
  </div>
  <img
    className="sc-htpNat iZvbbW"
    data-disabled={true}
    onClick={[Function]}
    src="/assets/img/common/left-arrow.png"
  />
  <img
    className="sc-bxivhb ebYeZP"
    onClick={[Function]}
    src="/assets/img/common/right-arrow.png"
  />
</div>
`;
